en/docs/develop/writing-a-custom-federated-authenticator.md

```diff
@@ -1,13 +1,18 @@
 # Writing a Custom Federated Authenticator
 
-It is possible to use the extension points available in the WSO2
-Identity Server to create custom federated authenticators.
+A custom federated authenticator can be written to authenticate a user with an external system. 
+So the external system can be any Identity provider such as Facebook, Twitter, Google and Yahoo. 
+It is possible to use the extension points available in the WSO2 Identity Server to create custom 
+federated authenticators.
+
+![Federated authentication diagram](../assets/img/using-wso2-identity-server/federated-authentication-diagram.png)
 
 ### Authenticator API
 
 The following is the API used to implement a custom federated
 authenticator.
 
+??? example "Click to view the API of the Application Authenticators"
 ``` java
```

Comment:
Does this appear properly in the local? The code should be a tab spacing away to fall under an expandable.

Reply:
Fixed https://github.com/wso2/docs-is/pull/2527/commits/ffc80dd6ac9088892baa7c56e5f03a9aa0768d3f